### ISSUE TITLE
sparks-and-fire compatibility issue with Apple M1

### ISSUE THREAD

**Issue by ionztorm** [hackr-sh/ghostty-shaders#24]
Have not yet tested other shaders, but so far the sparks-and-fire shader seems to have a compatibility issue withe the M1 silicone devices. 

It Ghostty loads with a white bg:
<img width="818" alt="Screenshot 2024-12-31 at 11 18 15" src="https://github.com/user-attachments/assets/631a5be5-d692-4353-aa09-35f032d180d9" />

**Comment by m-ahdal** [comment]
Can anyone with other Apple Silicon chips verify if this is a problem on their end too? I can verify I get the same issue on my M1 MAX MBP

**Comment by ionztorm** [comment]
> Can anyone with other Apple Silicon chips verify if this is a problem on their end too? I can verify I get the same issue on my M1 MAX MBP

Works fine on my M3 Max, and there are folk in the discord with no issues on M4.

I've a help thread open in the Ghostty discord where a few others on M1 have the same issue.

**Comment by ionztorm** [comment]
Someone in the discord got it to work somewhat by replacing `fract` with `smoothstep` in `noise1_2`

**Comment by ionztorm** [comment]
> If you swap the comment at the lines (https://github.com/m-ahdal/ghostty-shaders/blob/main/sparks-from-fire.glsl#L72), it works for me on M1Pro. Without using `smoothstep` it doesn't work and also displays a white blank terminal.
> 
> ```diff
> vec2 noise1_2(vec2 uv)
> {
> --    // vec2 f = fract(uv);
> ++    vec2 f = smoothstep(0.0, 1.0, fract(uv));
>     
>  	vec2 uv00 = floor(uv);
> ```

Yeah this is the change we tried - the only downside of course is that you lose the smoke effect

**Comment by linkarzu** [comment]
- I'm running an M1 mac mini, same white background, I reviewed shaders yesterday in this video
- [Ghostty Shaders | Ghostty config file syntax highlighting | glsl treesitter | glsl_analyzer](https://youtu.be/yJDn__NhOqI)
- I thought this was the desired effect, and thought to myself, "crazy what folks are into"

**Comment by joshuaauger** [comment]
None of the recommendations fix this issue on M2 Pro.

🤷🏻‍♂️

**Comment by hackr-sh** [comment]
Coming back to this - I'm going to go ahead and make the change above swapping the comment.
